cpp/include/cudf/strings/string.cuh

```diff
@@ -52,6 +52,40 @@ inline __device__ bool is_integer(string_view const& d_str)
            thrust::seq, begin, end, [] __device__(auto chr) { return chr >= '0' && chr <= '9'; });
 }
 
+/**
+ * @brief Returns true if input contains the not-a-number string.
+ *
+ * The following are valid for this function: "NAN" and "NaN"
+ * @param d_str input string
+ * @return true if input is as valid NaN string.
+ */
+inline __device__ bool is_nan_str(string_view const& d_str)
+{
+  auto const ptr = d_str.data();
+  return (d_str.size_bytes() == 3) && (ptr[0] == 'N' || ptr[0] == 'n') &&
+         (ptr[1] == 'A' || ptr[1] == 'a') && (ptr[2] == 'N' || ptr[2] == 'n');
```

Comment:
Something to consider for the future is that it feels like it would be useful to have a utility for comparing two `string_view`s for equality that ignores case like `strcasecmp`.

Reply:
Yep. This is non-trivial unless the text is all ASCII since it would require a couple of lookup tables. 
Regardless, an ignore-case compare function will be needed eventually for other strings requirements.

Reply:
I remember a trick that uses shifting ops to transform a character to lower-only or UPPER-ONLY case. Here it is: https://www.geeksforgeeks.org/case-conversion-lower-upper-vice-versa-string-using-bitwise-operators-cc/

Reply:
Yes, thanks. I knew about this trick but found out the optimizer knows it too so I coded it to be clearer for humans.
https://godbolt.org/z/dYh7s6vjd